extensions/amp-video/1.0/amp-video.js

```diff
@@ -0,0 +1,121 @@
+ * limitations under the License.
+ */
+
+import {ActionTrust} from '../../../src/action-constants';
+import {PreactBaseElement} from '../../../src/preact/base-element';
+import {VideoWrapper} from './video-wrapper';
+import {isLayoutSizeDefined} from '../../../src/layout';
+
+/** @const {string} */
+const TAG = 'amp-video';
+
+/**
+ * @param {PreactBaseElement<VideoDef.VideoApi>} impl
+ * @param {string} alias
+ * @param {function(!API_TYPE, !../service/action-impl.ActionInvocation)} handler
+ * @param {../action-constants.ActionTrust=} minTrust
+ */
+export function registerVideoAction(
+  impl,
+  alias,
+  handler,
+  minTrust = ActionTrust.HIGH
+) {
+  impl.registerApiAction(
+    alias,
+    (api, invocation) => {
+      if (invocation.trust >= ActionTrust.HIGH) {
+        api.userInteracted();
+      }
+      handler(api, invocation);
+    },
+    minTrust
+  );
+}
+
+/**
+ * @param {PreactBaseElement<VideoDef.VideoApi>} impl
+ */
+export function registerVideoActions(impl) {
+  registerVideoAction(impl, 'play', (api) => api.play());
+  registerVideoAction(impl, 'pause', (api) => api.pause());
+  registerVideoAction(impl, 'mute', (api) => api.mute());
+  registerVideoAction(impl, 'unmute', (api) => api.unmute());
+
+  // Ugly that this action has three names, but:
+  // - requestFullscreen for consistency with Element.requestFullscreen
+  // - fullscreenenter / fullscreen are for backwards compatibility
+  const requestFullscreen = (api) => api.requestFullscreen();
+  registerVideoAction(impl, 'requestFullscreen', requestFullscreen);
+  registerVideoAction(impl, 'fullscreenenter', requestFullscreen);
+  registerVideoAction(impl, 'fullscreen', requestFullscreen);
+}
+
+/** @extends {PreactBaseElement<VideoDef.VideoApi>} */
+class AmpVideo extends PreactBaseElement {
+  /** @override */
+  init() {
+    registerVideoActions(this);
+  }
+
+  /** @override */
+  isLayoutSupported(layout) {
+    return isLayoutSizeDefined(layout);
+  }
+}
+
+/** @override */
+AmpVideo['Component'] = VideoWrapper;
+
+/** @override */
+AmpVideo['props'] = {
+  'album': {attr: 'album'},
+  'alt': {attr: 'alt'},
+  'artist': {attr: 'artist'},
+  'artwork': {attr: 'artwork'},
+  'attribution': {attr: 'attribution'},
+  'autoplay': {attr: 'autoplay', type: 'boolean'},
+  'controls': {attr: 'controls', type: 'boolean'},
+  'controlslist': {attr: 'controlslist'},
+  'crossorigin': {attr: 'crossorigin'},
+  'disableremoteplayback': {attr: 'disableremoteplayback'},
+  'loop': {attr: 'loop', type: 'boolean'},
+  'noaudio': {attr: 'noaudio', type: 'boolean'},
+  'poster': {attr: 'poster'},
+  'src': {attr: 'src'},
+  'title': {attr: 'title'},
+
+  // TODO(alanorozco): These props have no internal implementation yet.
+  'dock': {attr: 'dock'},
+  'rotate-to-fullscreen': {attr: 'rotate-to-fullscreen', type: 'boolean'},
+};
+
+/** @override */
+AmpVideo['layoutSizeDefined'] = true;
+
+/** @override */
+AmpVideo['children'] = {
+  'sources': {
+    name: 'children',
```

Comment:
Should we also call it `sources`? Or do you actually like the idea of children? One concern I have is that our video player might get more children in the future. And in React it's very hard to differentiate between children.

Reply:
I like passing in as `children` since then `VideoWrapper` can use a `<video>` element directly, rather than an intermediate component that would use `sources` as children.

Reply:
I see. So you think "other" type of children are unlikely for any other player? This could be a somewhat of a forward compatibility pain in the future. Just calling it out. If you feel like we will be able to transition if needed - great - just let me know.

From Preact point of view, the difference is very small. It's basically `<video>{children}</video>` vs `<video>{sources}</video>`. So it's really only a question of API and what other elements might need.

Reply:
For now, using `sources` seems okay. Changed.